explorer/src/components/TokensCard.tsx

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+import classNames from "classnames";
+import { LoadingCard } from "./common/LoadingCard";
+import { ErrorCard } from "./common/ErrorCard";
+import { Sparkline, Direction } from "./Sparkline";
+import { normalizePercentage } from "utils";
+import {
+  useCoinGeckoCategoryTokens,
+  COIN_GECKO_SOLANA_CATEGORY,
+  CoinGeckoToken,
+} from "utils/coingecko";
+
+export function TokenRow({
+  token,
+  rank,
+}: {
+  token: CoinGeckoToken;
+  rank: number;
+}) {
+  return (
+    <tr>
+      <td>
+        <span className="badge badge-soft-gray badge-pill">{rank}</span>
+      </td>
+      <td className="">
+        <img className="token-icon mr-3" src={token.image} alt={token.name} />
+        {token.name}
+      </td>
+      <td className="text-muted">{token.symbol.toUpperCase()}</td>
+      <td className="text-right">
+        {token.current_price && "$" + token.current_price}
+      </td>
+      <td
+        className={classNames(
+          "text-right",
+          token.price_change_percentage_1h_in_currency > 0
+            ? "change-positive"
+            : "change-negative"
+        )}
+      >
+        {normalizePercentage(token.price_change_percentage_1h_in_currency, 2)}
+      </td>
+      <td
+        className={classNames(
+          "text-right",
+          token.price_change_percentage_1h_in_currency > 0
+            ? "change-positive"
+            : "change-negative"
+        )}
+      >
+        {normalizePercentage(token.price_change_percentage_24h, 2)}
+      </td>
+      <td
+        className={classNames(
+          "text-right",
+          token.price_change_percentage_1h_in_currency > 0
+            ? "change-positive"
+            : "change-negative"
+        )}
+      >
+        {normalizePercentage(token.price_change_percentage_7d_in_currency, 2)}
+      </td>
+      <td className="text-right">
+        {token.total_volume?.toLocaleString("en-US")}
+      </td>
+      <td className="text-right">
+        {token.market_cap?.toLocaleString("en-US")}
+      </td>
+      <td>
+        <Sparkline
+          values={token.sparkline_in_7d.price}
+          direction={
+            token.price_change_percentage_7d_in_currency > 0
+              ? Direction.Positive
+              : Direction.Negative
+          }
+        />
+      </td>
+    </tr>
+  );
+}
+
+export function TokensCard() {
+  const [error, loading, tokens] = useCoinGeckoCategoryTokens(
+    COIN_GECKO_SOLANA_CATEGORY
+  );
+
+  if (error) return <ErrorCard text={error.toString()} />;
+
+  if (loading) return <LoadingCard />;
```

Comment:
would you consider using Suspense for data fetching?

Reply:
Suspense would be a nice to have but I would skip it for now for two reasons 

- This is a change I would prefer to make app wide
- SSR for Suspense isn't ready yet and that _could_ be an issue if we decide that we'd like to run SSR in the future